bundles/org.openhab.binding.bigassfan/src/main/java/org/openhab/binding/bigassfan/internal/discovery/BigAssFanDiscoveryService.java

```diff
@@ -167,11 +173,11 @@ private synchronized void listen() {
                 break;
             }
         }
-        discoveryListener.shutdown();
+        localDiscoveryListener.shutdown();
         logger.debug("DiscoveryListener job is exiting");
     }
 
-    private void processMessage(BigAssFanDevice device) {
+    private void processMessage(@Nullable BigAssFanDevice device) {
```

Comment:
Perhaps it would be better to not call this method with a null argument, since the method won't do anything useful then?